apps/src/sites/studio/pages/teacher_dashboard/show.js

```diff
@@ -50,61 +46,41 @@ $(document).ready(function() {
   // TODO: (madelynkasula) remove duplication in sectionData.setSection and teacherSections.setSections
   store.dispatch(setSection(section));
   store.dispatch(setSections(sections));
-
   store.dispatch(selectSection(section.id));
   store.dispatch(setRosterProvider(section.login_type));
   store.dispatch(setLoginType(section.login_type));
-  store.dispatch(asyncSetCompletedLevelCount(section.id));
 
   if (!section.sharing_disabled && section.script.project_sharing) {
     store.dispatch(setShowSharingColumn(true));
   }
 
-  $.ajax({
-    method: 'GET',
-    url: `/dashboardapi/sections/${section.id}/students`,
-    dataType: 'json'
-  }).done(studentData => {
-    const convertedStudentData = convertStudentServerData(
-      studentData,
-      section.login_type,
-      section.id
-    );
-    store.dispatch(setStudents(convertedStudentData));
-  });
-
-  $.ajax({
-    method: 'GET',
-    url: '/dashboardapi/sections/valid_scripts',
-    dataType: 'json'
-  }).done(validScripts => {
-    store.dispatch(loadValidScripts(section, validScripts)).then(() => {
-      const scriptId = store.getState().scriptSelection.scriptId;
-      store.dispatch(asyncLoadTextResponses(section.id, scriptId));
-
-      renderTeacherDashboard();
-    });
-  });
+  store.dispatch(
+    setValidScripts(
+      validScripts,
+      studentScriptIds,
+      validCourses,
+      section.course_id
+    )
+  );
 
-  const renderTeacherDashboard = () => {
-    ReactDOM.render(
-      <Provider store={store}>
-        <Router basename={baseUrl}>
-          <Route
-            path="/"
-            component={props => (
-              <TeacherDashboard
-                {...props}
-                studioUrlPrefix={scriptData.studioUrlPrefix}
-                pegasusUrlPrefix={scriptData.pegasusUrlPrefix}
-                sectionId={section.id}
-                sectionName={section.name}
-              />
-            )}
-          />
-        </Router>
-      </Provider>,
-      document.getElementById('teacher-dashboard')
-    );
-  };
+  ReactDOM.render(
+    <Provider store={store}>
+      <Router basename={baseUrl}>
+        <Route
+          path="/"
+          component={props => (
+            <TeacherDashboard
+              {...props}
+              studioUrlPrefix={scriptData.studioUrlPrefix}
+              pegasusUrlPrefix={scriptData.pegasusUrlPrefix}
+              sectionId={section.id}
+              sectionName={section.name}
+              studentCount={section.students.length}
+            />
+          )}
+        />
+      </Router>
+    </Provider>,
```

Comment:
Thoughts on future work, nonblocking to this PR: Seems like we're using `react-router` in a nonobvious way here.  Might be a nice future cleanup to pull the routes up into this component (or push router down?) before we add dynamic loading.

Reply:
Good call Brad, I would expect to see the different react components listed within the `Router` block here as done in workshop dashboard: https://github.com/code-dot-org/code-dot-org/blob/1cad5ac667f9ed1a3aa31063e6d06bd21a78e2bd/apps/src/code-studio/pd/workshop_dashboard/workshop_dashboard.jsx#L85-L92

+1 for cleaning this up prior to implementing any performance optimizations related to lazy-loading.

Reply:
Added to backlog:  https://codedotorg.atlassian.net/browse/LP-841